pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -592,19 +659,42 @@ func (r *ReconcileBareMetalHost) manageHostPower(prov provisioner.Provisioner, i
 		return actionContinue{provResult.RequeueAfter}
 	}
 
+	pendingRebootSince := info.host.Status.Provisioning.PendingRebootSince
+	lastPoweredOn := info.host.Status.Provisioning.LastPoweredOn
+
+	desiredPowerOnState := info.host.Spec.Online
+	isInRebootProcess := lastPoweredOn.Before(pendingRebootSince) || (pendingRebootSince != nil && lastPoweredOn == nil)
+
+	if isInRebootProcess {
+		suffixlessAnnotationExists, shouldHoldPowerOff := getRebootAnnotations(info.host)
+
+		if info.host.Status.PoweredOn || shouldHoldPowerOff {
+			desiredPowerOnState = false
+		} else {
```

Comment:
I think this might be easier to understand as two separate blocks rather than if/else:

```
if isInRebootProcess {
    if desiredPowerOnState && !info.host.Status.PoweredOn {
        if _, hasAnn := info.host.Annotations[rebootAnnotationPrefix]; hasAnn {
            delete(info.host.Annotations, rebootAnnotationPrefix)
            return actionContinue{}
        }
    }
    if info.host.Status.PoweredOn || hasRebootAnnotation(info.host) {
       desiredPowerOnState = false
    }
}
```

(I don't think it's actually necessary, or even desirable, to check the desiredPowerOnState before removing the suffixless annotation, but in the spec we said we would.)

Reply:
Nice, I'll change the code accordingly.

> I don't think it's actually necessary, or even desirable, to check the desiredPowerOnState before removing the suffixless annotation, but in the spec we said we would

Perhaps we should change the spec?

Reply:
@zaneb does the spec become a design artefact at this point? Or do we keep it updated as the implementation changes?